### PR TITLE
Restore add/remove liquidity

### LINKED ISSUE #972: Restore Add/Remove liquidity
Not much more to explain.
Also take in consideration https://github.com/oceanprotocol/market/issues/894

Done:
- you can successfully add liquidity and the amount added is reflected in the pool stats without refreshing page
- you can successfully remove liquidity and the amount removed is reflected in the pool stats without refreshing page

**Blocked by #954**

### PR DESCRIPTION
**Pull request by kremalicious**
The just make it work part. As seen in #1014, there might be lots of other design & UI changes we need to do so numbers actually make sense.

* closes #972
* remove adding datatoken to pool, remove `CoinSelect` components and all logic for switching tokens
* remove the "Advanced" removal screen
* partially addresses #468 by reducing hardcoded OCEAN to `baseToken` in many places, which would allow reading out `baseToken` from some market config later on

### CODE REVIEW

File: src/components/@shared/TokenApproval/index.tsx [hunk 1]
```diff
@@ -29,34 +30,39 @@ export default function TokenApproval({
...
-    //   accountId,
-    //   spender
-    // )
-    // amount &&
-    //   new Decimal(amount).greaterThan(new Decimal('0')) &&
-    //   setTokenApproved(
-    //     new Decimal(allowance).greaterThanOrEqualTo(new Decimal(amount))
-    //   )
-  }, [tokenAddress, spender, accountId, amount, isAssetNetwork])
+  // TODO: how to check if token is approved as .allowance does not exist?
```
Comment (kremalicious) [inline src/components/@shared/TokenApproval/index.tsx:33]: Ideas for checking if token and amount have already been approved for user + pool are very welcome here.

File: src/components/Asset/AssetActions/Pool/Add/index.tsx [hunk 2]
```diff
@@ -80,38 +81,58 @@ export default function Add({
...
+    dtAddress,
+    coin,
+    dtBalance,
+    balance.ocean
+  ])
 
   // Submit
   async function handleAddLiquidity(amount: number, resetForm: () => void) {
+    const poolInstance = new Pool(web3, LoggerInstance)
+    const minPoolAmountOut = '0' // ? TODO: how to get?
```
Comment (kremalicious) [inline src/components/Asset/AssetActions/Pool/Add/index.tsx:125]: how to calculate this for `joinswapExternAmountIn()`? Doe we have ocean.js method or contract method for this or is this supposed to be calculated based on user input value combined with slippage value?

File: src/components/@shared/TokenApproval/index.tsx (hunk 1 above)
Comment (mihaisc) [reply src/components/@shared/TokenApproval/index.tsx:33]: fixed in new version of ocean.js

File: src/components/Asset/AssetActions/Pool/Add/index.tsx (hunk 2 above)
Comment (mihaisc) [reply src/components/Asset/AssetActions/Pool/Add/index.tsx:98]: this needs some thinking, i think fees should be taken in consideration as well

**Comment by kremalicious** [comment]
Should work™ as defined in linked issue and we need this here to better test further functionality like #1020 with which the updating liquidity after adding/removing should also work more reliably.

One thing already found, think the `.allowance()` doesn't work as expected, always returns `0` for me despite former approval. Also addd more logging in the PR, so that's more obvious now:

Screen Shot 2022-01-26 at 12 07 51

File: src/components/Asset/AssetActions/Pool/Add/index.tsx (hunk 2 above)
Comment (kremalicious) [reply src/components/Asset/AssetActions/Pool/Add/index.tsx:104]: good point, moved for now to https://github.com/oceanprotocol/market/issues/1027

**Pull request merged**